src/App.tsx

```diff
@@ -1,26 +1,44 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useDispatch } from 'react-redux';
+import { setTodos } from './features/todos';
+import { useEffect, useState } from 'react';
+import { getTodos } from './api';
+import { useAppSelector } from './app/hook';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const dispatch = useDispatch();
+  const [loading, setLoading] = useState<boolean>(false);
+  const currentTodo = useAppSelector(state => state.currentTodo);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(db => dispatch(setTodos(db)))
+      .finally(() => setLoading(false));
+  }, [currentTodo]);
```

Comment:
The `useEffect` hook is currently set to run whenever `currentTodo` changes. This might not be the intended behavior, as it will refetch todos every time `currentTodo` changes. Consider using an empty dependency array `[]` to fetch todos only once when the component mounts.